starknet_py/net/signer/base_signer.py

```diff
@@ -29,12 +30,22 @@ def sign_transaction(self, transaction: Transaction) -> List[int]:
         """
 
     @abstractmethod
-    def sign_message(self, typed_data: TypedData, account_address: int) -> List[int]:
+    def sign_message(
+        self, typed_data: Union[TypedData, TypedDataDataclass], account_address: int
+    ) -> List[int]:
         """
         Sign an TypedData TypedDict for off-chain usage with the starknet private key and return the signature
         This adds a message prefix, so it can't be interchanged with transactions
 
-        :param typed_data: TypedData TypedDict to be signed
+        :param typed_data:
+            TypedData TypedDict to be signed
+
+             .. deprecated:: 1.0.0
+                TypedData as dict has been deprecated as possible argument.
+                Use :py:class:`starknet_py.utils.TypedData` instead
         :param account_address: account address
         :return: the signature of the JSON object
+
+
         """
+        # TODO use proper deprecated version
```

Comment:
👀

Reply:
Cannot resolve until we decide what version this change will be (1.0.0 perhaps?)